packages/odk/src/Adaptor.js

```diff
@@ -0,0 +1,148 @@
+import * as util from './Utils';
+
+/**
+ * Execute a sequence of operations.
+ * Wraps `language-common/execute`, and prepends initial state for commcare.
+ * @example
+ * execute(
+ *   create('foo'),
+ *   delete('bar')
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(
+      util.authorize,
+      ...operations
+    )({ ...initialState, ...state });
+  };
+}
+
+/**
+ * Make a GET request
+ * @example
+ * get("v1/projects");
+ * @function
+ * @param {string} path - Path to resource
+ * @param {Object} params - Optional request params
+ * @param {Object} headers - Optional request headers
+ * @param {function} [callback] - Optional callback to handle the response
+ * @returns {Operation}
+ */
+export function get(path, params, headers, callback) {
+  return request('GET', path, null, params, callback, headers);
+}
+
+/**
+ * Make a GET request
+ * @example
+ * getForms(22);
+ * @function
+ * @param {number} id - Id of the project you want to get its forms
+ * @param {Object} params - Optional request params
+ * @param {Object} headers - Optional request headers
+ * @param {function} [callback] - Optional callback to handle the response
+ * @returns {Operation}
+ */
+export function getForms(id, params, headers, callback) {
+  const path = `/v1/projects/${id}/forms`;
+  return request('GET', path, null, params, callback, headers);
+}
+
+/**
+ * Make a GET request
+ * @example
+ * getSubmissions(22, 'test');
+ * @function
+ * @param {number} projectId - Id of the project you want to get its forms submissions
+ * @param {string} xmlFormId - Id of the form you want to get its submissions
+ * @param {Object} params - Optional request params
+ * @param {Object} headers - Optional request headers
+ * @param {function} [callback] - Optional callback to handle the response
+ * @returns {Operation}
+ */
+export function getSubmissions(
+  projectId,
+  xmlFormId,
+  params,
+  headers,
+  callback
+) {
+  const path = `/v1/projects/${projectId}/forms/${xmlFormId}/submissions`;
+  return request('GET', path, null, params, callback, headers);
+}
+
+/**
+ * Make a POST request
+ * @example
+ * post('v1/projects', { name: 'Project Name' });
+ * @function
+ * @param {string} path - Path to resource
+ * @param {object} data - Object which will be attached to the POST body
+ * @param {Object} params - Optional request params
+ * @param {function} [callback] - Optional callback to handle the response
+ * @returns {Operation}
+ */
+export function post(path, data, params, callback) {
+  return request('POST', path, data, params, callback);
+}
+
+/**
+ * Make a general HTTP request
+ * @example
+ * request("POST", 'v1/projects', { name: 'Project Name' });
+ * @function
```

Comment:
Ah we need to add `@public` here. That's my fault - can you also update the the template so that the default operations have `@public`?

Reply:
Should I also add `@public` to `get` and `post`?

Reply:
Ooh yes please, on all public operations! Both here and in the template